pkg/nsx/services/common/compare.go

```diff
@@ -0,0 +1,55 @@
+package common
+
+import (
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data/serializers/cleanjson"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+)
+
+var log = logger.Log
+
+type Comparable interface {
+	Key() string
+	Value() data.DataValue
+}
+
+func CompareResource(existing Comparable, expected Comparable) (isChanged bool, actual Comparable) {
```

Comment:
So you removed simplifyXXX and XXXCompare, but introduced XXXToComparable and ComparableToXXX, and the code line is increasing...Let me think if it can be abstracted further...

Reply:
Actually, according to this question https://stackoverflow.com/questions/28800672/how-to-add-new-methods-to-an-existing-type-in-go , there is no better way to extend existing struct.